internal/consul/namespaces.go

```diff
@@ -0,0 +1,50 @@
+package consul
+
+import (
+	capi "github.com/hashicorp/consul/api"
+)
+
+const (
+	WildcardNamespace = "*"
+	DefaultNamespace  = "default"
+)
+
+// EnsureNamespaceExists ensures a Consul namespace with name ns exists. If it doesn't,
+// it will create it and set crossNSACLPolicy as a policy default.
+// Boolean return value indicates if the namespace was created by this call.
+func EnsureNamespaceExists(client *capi.Client, ns string, crossNSAClPolicy string) (bool, error) {
```

Comment:
It seems like `crossNSACLPolicy` is only ever `""` in our codebase. Do we need to support it anyways, or could we drop the extra parameter for now and add back later if needed?